public/app/features/scenes/dashboard/DashboardSceneRenderer.tsx

```diff
@@ -1,85 +1,48 @@
 
 export function DashboardSceneRenderer({ model }: SceneComponentProps<DashboardScene>) {
-  const { title, body, actions = [], controls, isEditing, isDirty, uid } = model.useState();
+  const { body, controls, inspectPanelKey, viewPanelKey } = model.useState();
   const styles = useStyles2(getStyles);
-  const toolbarActions = (actions ?? []).map((action) => <action.Component key={action.state.key} model={action} />);
-
-  if (uid) {
-    toolbarActions.push(
-      <DashNavButton
-        key="button-scenes"
-        tooltip={'View as dashboard'}
-        icon="apps"
-        onClick={() => locationService.push(`/d/${uid}`)}
-      />
-    );
-  }
-
-  toolbarActions.push(<NavToolbarSeparator leftActionsSeparator />);
-
-  if (!isEditing) {
-    // TODO check permissions
-    toolbarActions.push(
-      <Button
-        onClick={model.onEnterEditMode}
-        tooltip="Enter edit mode"
-        key="edit"
-        variant="primary"
-        icon="pen"
-        fill="text"
-      >
-        Edit
-      </Button>
-    );
-  } else {
-    // TODO check permissions
-    toolbarActions.push(
-      <Button onClick={model.onEnterEditMode} tooltip="Save as copy" fill="text" key="save-as">
-        Save as
-      </Button>
-    );
-    toolbarActions.push(
-      <Button onClick={model.onDiscard} tooltip="Save changes" fill="text" key="discard" variant="destructive">
-        Discard
-      </Button>
-    );
-    toolbarActions.push(
-      <Button onClick={model.onEnterEditMode} tooltip="Save changes" key="save" disabled={!isDirty}>
-        Save
-      </Button>
-    );
-  }
+  const inspectPanel = model.findPanel(inspectPanelKey);
+  const viewPanel = model.findPanel(viewPanelKey);
+  const location = useLocation();
+  const pageNav = model.getPageNav(location);
 
   return (
-    <Page navId="scenes" pageNav={{ text: title }} layout={PageLayoutType.Custom}>
+    <Page navId="scenes" pageNav={pageNav} layout={PageLayoutType.Custom}>
       <CustomScrollbar autoHeightMin={'100%'}>
         <div className={styles.canvasContent}>
-          <AppChromeUpdate actions={toolbarActions} />
+          <NavToolbarActions dashboard={model} />
           {controls && (
             <div className={styles.controls}>
               {controls.map((control) => (
                 <control.Component key={control.state.key} model={control} />
               ))}
             </div>
           )}
-          <div className={styles.body}>
-            <body.Component model={body} />
-          </div>
+          {viewPanel ? (
+            <div className={styles.viewPanel}>
+              <viewPanel.Component model={viewPanel} />
+            </div>
+          ) : (
+            <div className={styles.body}>
+              <body.Component model={body} />
+            </div>
+          )}
         </div>
       </CustomScrollbar>
+      {inspectPanel && <ScenePanelInspector panel={inspectPanel} dashboard={model} />}
```

Comment:
I like the simplicity of this, but feels like something hard to port as a declarative object in a scene.

Reply:
@dprokop not sure I understand? Do you think the panel inspect should be a SceneObject + Renderer (Component), instead of just a plain react component? 

I think that could very good idea (depending on the inspector implementation and the possibility to share code with the current)

Reply:
Have not put too much thought into the inspector yet, but yes, I was thinking about a declarative inspector. The problem that we gonna face is the tabs customization tho. I don't think all of the current inspection will be applicable in scenes - like data preview. So Maybe we should have a primitive SceneDataInspect that allows tabs configuration - which would be other, more specifig scene objects? The standard ones I can think about are the Data and Stats. The JSON, Query and Stats will probably be different for apps and for Dashboards.